### ISSUE TITLE
Implement sanitation module

### ISSUE THREAD

**Issue by ClaudiaComito** [helmholtz-analytics/heat#468]
**Feature functionality**
Collect sanitation checks in a dedicated module. This would include the omnipresent 

```
if not isinstance(x, dndarray.DNDarray):
        raise TypeError("expected x to be a ht.DNDarray, but was {}".format(type(x)))
```
and variations thereof, but also more specific checks needed in high-level algorithms. See sklearn's validation module:

https://github.com/scikit-learn/scikit-learn/blob/b194674c4/sklearn/utils/validation.py

**Comment by Cdebus** [comment]
This should probably also handle communication and device sanitation. That is something that pops up frequently, but is a bit tedious to check for in each function individually. So if you make a general sanitation function, could you include this?

**Comment by ClaudiaComito** [comment]
Closed by #662
